hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/ScheduleSigTransactionMetadataBuilder.java

```diff
@@ -50,12 +53,21 @@ public ScheduleTransactionMetadata build() {
         Objects.requireNonNull(
                 txn, "Transaction body is required to build ScheduleSigTransactionMetadata");
         Objects.requireNonNull(payer, "Payer is required to build ScheduleSigTransactionMetadata");
+        if (status == ResponseCodeEnum.INVALID_SCHEDULE_ID) {
+            return new InvalidTransactionMetadata(txn, payer, status);
+        }
```

Comment:
Why do we need this check separately ? Since we have already computed all needed fields we could return the `ScheduleSigTransactionMetadata`